field_manager/fm_yaml.F90

```diff
@@ -31,493 +31,320 @@
-!> @ingroup fm_yaml_mod
+!> This type contains the name of the field type, the block id, and the models within this field type
 type, public :: fmType_t
-  integer                       :: yfid                !< file id of a yaml file
   integer                       :: id                  !< block id of this type
   character(len=:), allocatable :: name                !< name of the type
-  character(len=7)              :: blockname="modlist" !< name of the root block
-  integer                       :: nchildren           !< number of model types
-  integer, allocatable          :: child_ids(:)        !< array of model ids
-  type (fmModel_t), allocatable :: children(:)         !< models in this type
-  contains
-    procedure :: get_blocks => get_blocks_fmType_t
-    procedure :: destruct => destruct_fmType_t
-    procedure :: get_name => get_name_fmType_t
-    procedure :: create_children => create_children_fmType_t
+  type (fmModel_t), allocatable :: models(:)           !< models in this type
 end type fmType_t
 
-!> @brief This type represents the entirety of the field table.
-!> This type contains the file id of the yaml file, the field types within this table, and the following methods:
-!! getting blocks, creating children (field types), and self destruction.
-!> @ingroup fm_yaml_mod
+!> @brief This type contains the field types within a field table.
 type, public :: fmTable_t
-  integer                       :: yfid                    !< file id of a yaml file
-  character(len=11)             :: blockname="field_table" !< name of the root block
-  integer                       :: nchildren               !< number of field types
-  integer, allocatable          :: child_ids(:)            !< array of type ids
-  type (fmType_t), allocatable  :: children(:)             !< field types in this table
-  contains
-    procedure :: get_blocks => get_blocks_fmTable_t
-    procedure :: destruct => destruct_fmTable_t
-    procedure :: create_children => create_children_fmTable_t
+  type (fmType_t), allocatable :: types(:) !< field types in this table
 end type fmTable_t
 
-!> @brief Interface to construct the fmTable type.
-!> @ingroup fm_yaml_mod
-interface fmTable_t
-  module procedure construct_fmTable_t
-end interface fmTable_t
-
-!> @brief Interface to construct the fmType type.
-!> @ingroup fm_yaml_mod
-interface fmType_t
-  module procedure construct_fmType_t
-end interface fmType_t
-
-!> @brief Interface to construct the fmModel type.
-!> @ingroup fm_yaml_mod
-interface fmModel_t
-  module procedure construct_fmModel_t
-end interface fmModel_t
-
-!> @brief Interface to construct the fmVar type.
-!> @ingroup fm_yaml_mod
-interface fmVar_t
-  module procedure construct_fmVar_t
-end interface fmVar_t
-
-!> @brief Interface to construct the fmAttr type.
-!> @ingroup fm_yaml_mod
-interface fmAttr_t
-  module procedure construct_fmAttr_t
-end interface fmAttr_t
-
 contains
 
-!> @addtogroup fm_yaml_mod
-!> @{
```

Comment:
These two lines need to be added back in as well, it adds the routines to the module page.

Reply:
Added back in 02085437c8de6c6d895c878158f2186edf07eacd.